ext/drbbdup/drbbdup.c

```diff
@@ -562,28 +562,80 @@ drbbdup_do_orig_analysis(drbbdup_manager_t *manager, void *drcontext, void *tag,
     return orig_analysis_data;
 }
 
+static instr_t *
+drbbdup_get_next_where(instrlist_t *bb, IN instr_t *where)
+{
+    if (where == NULL)
+        return NULL;
+
+    instr_t *next_where = instr_get_next(where);
+    instr_t *last = instrlist_last_app(bb);
+    bool is_last_special = drbbdup_is_special_instr(last);
+
+    if (where == NULL || drbbdup_is_at_end(where) ||
+        (drbbdup_is_at_end(next_where) && !is_last_special))
```

Comment:
Please add a comment explaining: I'm not following why this needs to check for the current *and* the next being the end label.